### ISSUE TITLE
How to add a custom infix to existing transients

### ISSUE THREAD

**Issue by kai2nenobu** [magit/transient#14]
For example, I want to add `--first-parent` switch to `magit-log` transient.

[Document](https://magit.vc/manual/transient.html#Modifying-Existing-Transients) explains how to modify existing transients, but this is limited to modifying suffix.

Version Info

```
Emacs => GNU Emacs 26.1 (build 1, x86_64-w64-mingw32)
transient.el => 20190215.1221
magit.el => 20190215.1359

**Comment by kai2nenobu** [comment]
Sorry, I noticed that `magit-log` transient already has `--first-parent` infix, but it's only an example.  I want to know how to add a custom infix which is not defined by `define-transient-command`.

**Comment by kyleam** [comment]
>  I want to know how to add a custom infix which is not defined by `define-transient-command`.

See [here][0] and [here][1].

[0]: https://magit.vc/manual/transient/Modifying-Existing-Transients.html#Modifying-Existing-Transients
[1]: https://github.com/magit/magit/wiki/Converting-popup-modifications-to-transient-modifications

**Comment by kai2nenobu** [comment]
Thanks @kyleam.

I assumed that `transient-append-suffix` does not work with infix from its name.  The following code does work as I expected!

``` emacs-lisp
(transient-append-suffix 'magit-log "=m"
  '("-R" "Remove Empty" "--remove-empty"))
```

BTW, I noticed two mistakes.

Firstly, Doumentation about `SUFFIX` parameter of `transient-*-suffix` functions differ between DOCSTRING and HTML document (your first reference).

`DOCSTRING of transient-*-suffix`

>SUFFIX is a suffix command specification list of the same form as expected by ‘define-transient-command’.

`HTML document`

>SUFFIX is a transient infix or suffix specification in the same form as expected by define-transient-command.


HTML document refers infix, but DOCSTRING doesn't. It's nice both documentations explain as same.

Secondly, a magit wiki example (your second reference) in `Adding a switch` section doesn't work.

``` emacs-lisp
(transient-append-suffix 'magit-log "-A"
  '("-1" "First parent" "--first-parent"))
```

When I evaluated above code, error has occued as below.


```
Cannot insert (\"-1\" \"First parent\" \"--first-parent\") into magit-log; -A not found
```

If I specify `LOC` by command name instead of key, it does work.

``` emacs-lisp
(transient-append-suffix 'magit-log #'magit:--author
  '("-1" "First parent" "--first-parent"))
```

Thank you again!.

**Comment by tarsius** [comment]
> a magit wiki example (your second reference) in `Adding a switch`
> section doesn't work.

I've fixed that.

> [infix vs. suffix]

An infix is a special kind of suffix.  Depending on context "suffixes" means "suffixes (including infixes)" or "non-infix suffixes".  Earlier drafts of the documentation repeated that piece of information all over the place, but I removed many mentions.  I'll have to go over this again some time and re-add a note about that in certain places.
